osu.Game/Screens/Backgrounds/BackgroundScreenBeatmap.cs

```diff
@@ -112,5 +117,62 @@ private void load(TextureStore textures)
                 Sprite.Texture = Beatmap?.Background ?? textures.Get(@"Backgrounds/bg1");
             }
         }
+
+        public class DimmableBackground : UserDimContainer
+        {
+            /// <summary>
+            /// The amount of blur to be applied to the background in addition to user-specified blur.
+            /// </summary>
+            /// <remarks>
+            /// Used in contexts where there can potentially be both user and screen-specified blurring occuring at the same time, such as in <see cref="PlayerLoader"/>
+            /// </remarks>
+            public readonly Bindable<float> BlurAmount = new Bindable<float>();
+
+            public Background Background
+            {
+                get => background;
+                set
+                {
+                    base.Add(background = value);
+                    background.BlurTo(blurTarget, 0, Easing.OutQuint);
+                }
+            }
+
+            private Bindable<double> userBlurLevel { get; set; }
+
+            private Background background;
+
+            public override void Add(Drawable drawable)
+            {
+                if (drawable is Background)
+                    throw new InvalidOperationException($"Use {nameof(Background)} to set a background.");
+
+                base.Add(drawable);
+            }
+
+            /// <summary>
+            /// As an optimisation, we add the two blur portions to be applied rather than actually applying two separate blurs.
+            /// </summary>
+            private Vector2 blurTarget => EnableUserDim.Value
+                ? new Vector2(BlurAmount.Value + (float)userBlurLevel.Value * 25)
```

Comment:
Where'd the 25 come from?

Reply:
Again, not part of this PR. Just a constant which gave the user blur enough blur. Can move to a constant if it makes you feel better.

Reply:
I try to avoid magic numbers if I can, but if this is the only place it's used it's probably fine.